### DISCUSSION TITLE
Flattening Collections

### DISCUSSION THREAD

**Discussion by joaooo** [riok/mapperly#1097] [Q&A]
Is there a way i can achieve collections flattening without using beforeMap/afterMap/having to do my own implemention?
Example:

```
public record ProductDto
{
    public int Id { get; set; }
    public List<Version> Versions { get; set; }
}

public record Version
{
    public int Id { get; set; }
    public List<Multimedia> Multimedias { get; set; }
}
  
public record Multimedia
{
  public int Id { get; set; }
  public string Url { get; set; }
}
  
public record ProductResponse
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string ImageUrl { get; set; }
}

[MapProperty($"{nameof(ProductDto.Versions)}.{nameof(Version.Multimedias)}.{nameof(Multimedia.Url)}", nameof(ProductResponse.ImageUrl))]
    private partial ProductResponse Map(ProductDto source);
```

In this example my idea was to map the first/last/whatever, one, Multimedia.Url to my ImageUrl prop.
The MapProperty generates an obvious error, which is "specified member xxx on source type xxx was not found"

Thanks in advance

**Comment by TimothyMakkison** [comment]
Hey, unfortunately Mapperly doesn't support this. A before/after mapper should easily solve your issue 👍

**Comment by latonz** [comment]
I'm not really sure if I understand correctly. You want to map `ProductDto.Versions[*]. Multimedias[*].Url` to a single `ProductResponse.ImageUrl`. How do you map n*m entries to a single one? How would you expect the generated code to look like?

**Reply by joaooo** [reply]
Not sure if it would be possible but if we could implement a method that has that type of logic and knows how to map from N entries to one, then mapperly could use that

**Reply by latonz** [reply]
What do you expect the result to look like? Joined strings?

**Reply by joaooo** [reply]
No, and i understand this is kinda tricky because, we might want a different result, like you said yourself, maybe i would actually it to be joined strings. I was trying to think of a way to make it flexible, so the user could specify the TSource and UResult. In my specific case, it would be nice if mapperly forced me to implement a method that would map from a list of N and return what i wanted, in this case, a string. This way i wouldn't need to implement most of my methods, only their very specific mappings while mapperly would call them by itself

**Reply by latonz** [reply]
Closing for now as this seems to be out of scope for Mapperly.
